File: Parcial Paradigmas/Tercer Punto/IMPLEMENTACIONPREVIATRADUCCION/src/main/java/com/app/calculadora/cientifica/clases/Calculadora.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.app.calculadora.cientifica.clases;

import com.app.calculadora.cientifica.excepciones.EntradaInvalidaException;

public class Calculadora {

    private double valor;

    // Constructor para números enteros
    public Calculadora(int valor) {
        this.valor = valor;
    }

    // Constructor para números decimales
    public Calculadora(double valor) {
        this.valor = valor;
    }

    public Calculadora suma(Calculadora numero2) {
        return new Calculadora(this.valor + numero2.valor);
    }

    public Calculadora resta(Calculadora numero2) {
        return new Calculadora(this.valor - numero2.valor);
    }

    public Calculadora multiplicacion(Calculadora numero2) {
        return new Calculadora(this.valor * numero2.valor);
    }

    public Calculadora division(Calculadora numero2) throws EntradaInvalidaException {
        if (this.valor == 0 || numero2.valor == 0) {
            throw new EntradaInvalidaException("Error: No se puede dividir entre cero.");
        }
        return new Calculadora(this.valor / numero2.valor);
    }

    public double getValor() {
        return valor;
    }
}

File: Parcial Paradigmas/Tercer Punto TT/TercerPunto/src/Calculadora.kt
package com.app.calculadora.cientifica.clases

import com.app.calculadora.cientifica.excepciones.EntradaInvalidaException

// Clase base Calculadora
open class Calculadora(private val valor: Double) {

    // Operación de suma
    fun suma(otro: Calculadora): Calculadora {
        return Calculadora(this.valor + otro.valor)
    }

    // Operación de resta
    fun resta(otro: Calculadora): Calculadora {
        return Calculadora(this.valor - otro.valor)
    }

    // Operación de multiplicación
    fun multiplicacion(otro: Calculadora): Calculadora {
        return Calculadora(this.valor * otro.valor)
    }

    // Operación de división con manejo de excepción
    @Throws(EntradaInvalidaException::class)
    fun division(otro: Calculadora): Calculadora {
        if (otro.valor == 0.0) {
            throw EntradaInvalidaException("Error: No se puede dividir entre cero.")
        }
        return Calculadora(this.valor / otro.valor)
    }

    // Método para obtener el valor actual
    fun obtenerValor(): Double {
        return valor
    }
}